cmake/fake_tribits.cmake

```diff
@@ -185,8 +185,12 @@ function(KOKKOS_ADD_TEST_LIBRARY NAME)
 endfunction()
 
 function(KOKKOS_INCLUDE_DIRECTORIES)
-  cmake_parse_arguments(INC "REQUIRED_DURING_INSTALLATION_TESTING" "" "" ${ARGN})
-  include_directories(${INC_UNPARSED_ARGUMENTS})
+  cmake_parse_arguments(INC "REQUIRED_DURING_INSTALLATION_TESTING;SYSTEM" "" "" ${ARGN})
```

Comment:
1. `kokkos_include_directories` seems to be a (useless?) wrapper for `include_directories`.
2. Using `include_directories` is a bad idea, it goes against `CMake` target principles.
3. `REQUIRED_DURING_INSTALLATION_TESTING` seems a dead option.

Reply:
In the end, I don't need to modify this function but still, it seems useless. @masterleinad What do you think ? Is it for some `TriBits` compatibility reason ?

Reply:
The point of removing the special path for `Trilinos`/`TriBITS` builds was to be able to start cleaning up our `CMake` files. There are only a couple of lines left that we need for `TriBITS` interoperability but this shouldn't be one of them.

Reply:
OK, thanks for the insight :+1: